eos-paygd/main.c

```diff
@@ -38,14 +41,58 @@ sync_and_poweroff (gpointer user_data)
     return FALSE;
 }
 
+static void
+allow_writing_to_mmcblkxboot0 (gboolean allow_write)
+{
+  glob_t globbuf = { 0 };
+  int ret = glob ("/dev/mmcblk*boot0", 0, NULL, &globbuf);
```

Comment:
Split-disk Endless Ones have `/dev/mmcblk0` and `/dev/mmcblk1` but only one has a boot partition (`/dev/mmcblk0boot0`). I'd like to be able to assume that the boot partition is always `/dev/mmcblk0boot0` but that's not what's implied by the other code that deals with it:

- https://github.com/endlessm/eos-payg-nonfree/blob/0ad448aa566ed1a4835fd3bc6d3f48e94aa39bd8/eos-payg-angaza-provision/factory-test/start.sh#L33
 - https://github.com/endlessm/eos-payg-nonfree/blob/0ad448aa566ed1a4835fd3bc6d3f48e94aa39bd8/eos-payg-angaza-provision/eos-payg-angaza-unprovision-emmc#L28
 - https://github.com/endlessm/eos-payg-nonfree/blob/b6915e5a082642241708697baa1f8f5196c41b50/libeos-payg-angaza/nvm-backend-emmc.c#L362

The one exception is `eos-paygd-emmc.service` which had `/dev/mmcblk0boot0` hardcoded.

Reply:
Iirc Asus E406MA is the counterexample. However as you have noticed I ended up hardcoding the 0 path (I hit some issues where writes to the Asus device would not "stick" so gave up on  trying to support anything but the path in the EF20EA)

Reply:
So are we guaranteed that on devices which have only one `boot0` and it shows up under `/dev/mmcblk0` once it will always be numbered that way and never under `/dev/mmcblk1boot0`?

It's hard to see a good reason to rewrite this to only support `/dev/mmcblk0boot0`. Perhaps we'll need to support something like the E406MA in the future.

Reply:
The kernel makes no guarantees that mmcblk0 continues to be "0" on subsequent boots or on different kernel versions. Please avoid any reliance on that particular number.

Reply:
Okay. That still leaves the question of whether more than one eMMC device will have a boot0 partition. I don't know why that would ever happen but I think we should leave it as is unless we're certain.

Reply:
On a Hack computer (Asus E406M) I'm seeing only one boot0 partition:
```
$ ls /dev/mmcblk*
/dev/mmcblk0  /dev/mmcblk0boot0  /dev/mmcblk0boot1  /dev/mmcblk0p1  /dev/mmcblk0p2  /dev/mmcblk0p3  /dev/mmcblk0rpmb
```
So we should only use `/dev/mmcblk0boot0` in that case. I'll resolve this discussion; I think what we have will work for the foreseeable future.